app/src/main/java/org/mozilla/focus/fragment/DownloadDialogFragment.java

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.fragment;
+
+import android.os.Build;
+import android.os.Bundle;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+import android.text.Html;
+import android.text.Spanned;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.webkit.URLUtil;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.mozilla.focus.R;
+import org.mozilla.focus.telemetry.TelemetryWrapper;
+import org.mozilla.focus.web.Download;
+
+/**
+ * Fragment displaying a download dialog
+ */
+public class DownloadDialogFragment extends DialogFragment {
+
+    public static DownloadDialogFragment newInstance(Download download) {
+        DownloadDialogFragment frag = new DownloadDialogFragment();
+        final Bundle args = new Bundle();
+
+        String fileName = URLUtil.guessFileName(
+                download.getUrl(), download.getContentDisposition(), download.getMimeType());
+
+        args.putString("fileName", fileName);
+        args.putParcelable("download", download);
+        frag.setArguments(args);
+        return frag;
+    }
+
+    public AlertDialog onCreateDialog(Bundle bundle) {
```

Comment:
super-nit: fwiw, this function could benefit from helper functions (it's more than a screen long!) but no need to change it now and potentially reintroduce bugs (though, I guess intellij can actually do this refactor for you so... :P)